test/controllers/notifications_controller_test.rb

```diff
@@ -239,6 +239,23 @@ class NotificationsControllerTest < ActionDispatch::IntegrationTest
     assert_response :ok
   end
 
+  test 'gracefully handles failed user notification syncs' do
+    sign_in_as(@user)
+    User.any_instance.stubs(:sync_notifications).raises(Octokit::BadGateway)
```

Comment:
The use of `any_instance` here makes me cringe, but alas it's how we're doing other such specs.

We should refactor the test helper a bit and see about allowing an actual user object to be passed to avoid this but leaving it for now.